java/operator/src/main/java/com/instaclustr/cassandra/operator/controller/DataCenterReconciliationController.java

```diff
@@ -0,0 +1,546 @@
+            config.put("seed_provider", ImmutableList.of(ImmutableMap.of(
+                            "class_name", "com.instaclustr.cassandra.k8s.SeedProvider",
+                            "parameters", ImmutableList.of(ImmutableMap.of("service", seedNodesService.getMetadata().getName()))
+                    )));
+
+
+//            config.put("endpoint_snitch", "com.instaclustr.cassandra.k8s.Snitch") // TODO: custom snitch, so that we don't need to write the config for GPFS
+            config.put("endpoint_snitch", "org.apache.cassandra.locator.GossipingPropertyFileSnitch");
+
+
+            configMapVolumeAddFile(configMap, volumeSource, "cassandra.yaml.d/001-operator-overrides.yaml", toYamlString(config));
+        }
+
+        // GossipingPropertyFileSnitch config
+        {
+            final Properties rackDcProperties = new Properties();
+
+            rackDcProperties.setProperty("dc", dataCenterMetadata.getName());
+            rackDcProperties.setProperty("rack", "the-rack"); // TODO: support multiple racks
+            rackDcProperties.setProperty("prefer_local", "true"); // TODO: support multiple racks
+
+            final StringWriter writer = new StringWriter();
+            rackDcProperties.store(writer, "generated by cassandra-operator");
+
+            configMapVolumeAddFile(configMap, volumeSource, "cassandra-rackdc.properties", writer.toString());
+        }
+
+        // prometheus support
+        if (dataCenterSpec.getPrometheusSupport()) {
+            configMapVolumeAddFile(configMap, volumeSource, "cassandra-env.sh.d/001-cassandra-exporter.sh",
+                    "JVM_OPTS=\"${JVM_OPTS} -javaagent:${CASSANDRA_HOME}/agents/cassandra-exporter-agent.jar=@${CASSANDRA_CONF}/cassandra-exporter.conf\"");
+        }
+
+        // heap size and GC settings
+        // TODO: tune
+        {
+            final long coreCount = 4; // TODO: not hard-coded
+            final long memoryLimit = dataCenterSpec.getResources().getLimits().get("memory").getNumber().longValue();
+
+            // same as stock cassandra-env.sh
+            final long jvmHeapSize = Math.max(
+                    Math.min(memoryLimit / 2, 1 * GB),
+                    Math.min(memoryLimit / 4, 8 * GB)
+            );
+
+            final long youngGenSize = Math.min(
+                    MB * coreCount,
+                    jvmHeapSize / 4
+            );
+
+            final boolean useG1GC = (jvmHeapSize > 8 * GB);
+
+            final StringWriter writer = new StringWriter();
+            try (final PrintWriter printer = new PrintWriter(writer)) {
+                printer.format("-Xms%d%n", jvmHeapSize); // min heap size
+                printer.format("-Xmx%d%n", jvmHeapSize); // max heap size
+
+                // copied from stock jvm.options
+                if (!useG1GC) {
+                    printer.format("-Xmn%d%n", youngGenSize); // young gen size
```

Comment:
Can we make this a separate file or resource?

One approach would be a config map containing defaults for the operator, the operator reads this default from a path, but it's customizable on the installation of the operator (this can also be handled in helm installation chart for the operator).

Reply:
> Can we make this a separate file or resource?

As in a separate config-map file, or something on the operator side?

What would the intended outcome be? So that the GC settings can be customized by the user? For that, the user config map can override `jvm.options` or add additional files into `jvm.options.d` to alter or change any settings.